toolchain/check/eval_inst.cpp

```diff
@@ -102,6 +103,23 @@ auto EvalConstantInst(Context& context, SemIR::BindAlias inst)
       context.constant_values().Get(inst.value_id));
 }
 
+auto EvalConstantInst(Context& context, SemIR::BindName inst)
+    -> ConstantEvalResult {
+  if (!inst.value_id.has_value()) {
+    return ConstantEvalResult::NotConstant;
+  }
+
+  // A reference binding evaluates to the value it's bound to.
+  if (SemIR::IsReferenceCategory(
```

Comment:
Maybe drop the separate `if`? It's not clear to me why it's distinct (absence of comment not helping)
```suggestion
  // A reference binding evaluates to the value it's bound to.
  if (inst.value_id.has_value() &&
      SemIR::IsReferenceCategory(
```